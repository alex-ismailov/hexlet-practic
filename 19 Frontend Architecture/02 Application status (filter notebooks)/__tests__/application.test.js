import fs from 'fs';
import path from 'path';
import prettier from 'prettier';
import keycode from 'keycode';

import run from '../src/application.js';

const options = {
  parser: 'html',
  htmlWhitespaceSensitivity: 'ignore',
  tabWidth: 4,
};

const fixturesPath = path.join(__dirname, '__fixtures__');
const getTree = () => prettier.format(document.body.innerHTML, options);

const pressKey = (key, el = document.body, value = key) => {
  const keyCode = keycode(key);
  const e = new KeyboardEvent('input', { keyCode, bubbles: true });
  el.value = value; // eslint-disable-line
  el.setAttribute('value', value);
  el.dispatchEvent(e);
};

beforeEach(() => {
  const initHtml = fs.readFileSync(path.join(fixturesPath, 'index.html')).toString();
  document.documentElement.innerHTML = initHtml;
  run();
});

test('application', async () => {
  expect(getTree()).toMatchSnapshot();

  const frequencyMin = document.querySelector('input[name="frequency_gte"]');
  frequencyMin.focus();
  pressKey('3', frequencyMin);
  expect(getTree()).toMatchSnapshot();

  pressKey('Backspace', frequencyMin, '');
  expect(getTree()).toMatchSnapshot();

  pressKey('4', frequencyMin);
  expect(getTree()).toMatchSnapshot();

  pressKey('1', frequencyMin);
  expect(getTree()).toMatchSnapshot();

  const frequencyMax = document.querySelector('input[name="frequency_lte"]');
  pressKey('Backspace', frequencyMax, '');
  expect(getTree()).toMatchSnapshot();

  pressKey('2', frequencyMax);
  expect(getTree()).toMatchSnapshot();
});

test('application 2', async () => {
  expect(getTree()).toMatchSnapshot();
});
