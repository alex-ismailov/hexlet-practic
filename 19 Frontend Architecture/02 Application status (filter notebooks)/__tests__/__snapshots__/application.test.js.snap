// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application 1`] = `
"<div class=\\"container m-3\\">
    <div class=\\"row\\">
        <div class=\\"col-6\\">
            <form>
                <div>
                    <label>Processor</label>
                    <select name=\\"processor_eq\\">
                        <option value=\\"\\">Select</option>
                        <option value=\\"intel\\">Intel</option>
                        <option value=\\"amd\\">AMD</option>
                    </select>
                </div>
                <div>
                    <label>Memory</label>
                    <select name=\\"memory_eq\\">
                        <option value=\\"\\">Select</option>
                        <option value=\\"8\\">8</option>
                        <option value=\\"16\\">16</option>
                        <option value=\\"32\\">32</option>
                    </select>
                </div>
                <div>
                    <label>Frequency Min</label>
                    <input
                        type=\\"number\\"
                        name=\\"frequency_gte\\"
                        min=\\"1\\"
                        max=\\"5\\"
                        value=\\"\\"
                    />
                </div>
                <div>
                    <label>Frequency Max</label>
                    <input
                        type=\\"number\\"
                        name=\\"frequency_lte\\"
                        min=\\"1\\"
                        max=\\"5\\"
                        value=\\"\\"
                    />
                </div>
            </form>
        </div>
        <div class=\\"col-6\\">
            <div class=\\"result\\">
                <ul>
                    <li>v1</li>
                    <li>d3</li>
                    <li>d2</li>
                </ul>
            </div>
        </div>
    </div>
</div>
"
`;

exports[`application 2`] = `
"<div class=\\"container m-3\\">
    <div class=\\"row\\">
        <div class=\\"col-6\\">
            <form>
                <div>
                    <label>Processor</label>
                    <select name=\\"processor_eq\\">
                        <option value=\\"\\">Select</option>
                        <option value=\\"intel\\">Intel</option>
                        <option value=\\"amd\\">AMD</option>
                    </select>
                </div>
                <div>
                    <label>Memory</label>
                    <select name=\\"memory_eq\\">
                        <option value=\\"\\">Select</option>
                        <option value=\\"8\\">8</option>
                        <option value=\\"16\\">16</option>
                        <option value=\\"32\\">32</option>
                    </select>
                </div>
                <div>
                    <label>Frequency Min</label>
                    <input
                        type=\\"number\\"
                        name=\\"frequency_gte\\"
                        min=\\"1\\"
                        max=\\"5\\"
                        value=\\"3\\"
                    />
                </div>
                <div>
                    <label>Frequency Max</label>
                    <input
                        type=\\"number\\"
                        name=\\"frequency_lte\\"
                        min=\\"1\\"
                        max=\\"5\\"
                        value=\\"\\"
                    />
                </div>
            </form>
        </div>
        <div class=\\"col-6\\">
            <div class=\\"result\\">
                <ul>
                    <li>d3</li>
                </ul>
            </div>
        </div>
    </div>
</div>
"
`;

exports[`application 2 1`] = `
"<div class=\\"container m-3\\">
    <div class=\\"row\\">
        <div class=\\"col-6\\">
            <form>
                <div>
                    <label>Processor</label>
                    <select name=\\"processor_eq\\">
                        <option value=\\"\\">Select</option>
                        <option value=\\"intel\\">Intel</option>
                        <option value=\\"amd\\">AMD</option>
                    </select>
                </div>
                <div>
                    <label>Memory</label>
                    <select name=\\"memory_eq\\">
                        <option value=\\"\\">Select</option>
                        <option value=\\"8\\">8</option>
                        <option value=\\"16\\">16</option>
                        <option value=\\"32\\">32</option>
                    </select>
                </div>
                <div>
                    <label>Frequency Min</label>
                    <input
                        type=\\"number\\"
                        name=\\"frequency_gte\\"
                        min=\\"1\\"
                        max=\\"5\\"
                        value=\\"\\"
                    />
                </div>
                <div>
                    <label>Frequency Max</label>
                    <input
                        type=\\"number\\"
                        name=\\"frequency_lte\\"
                        min=\\"1\\"
                        max=\\"5\\"
                        value=\\"\\"
                    />
                </div>
            </form>
        </div>
        <div class=\\"col-6\\">
            <div class=\\"result\\">
                <ul>
                    <li>v1</li>
                    <li>d3</li>
                    <li>d2</li>
                </ul>
            </div>
        </div>
    </div>
</div>
"
`;

exports[`application 3`] = `
"<div class=\\"container m-3\\">
    <div class=\\"row\\">
        <div class=\\"col-6\\">
            <form>
                <div>
                    <label>Processor</label>
                    <select name=\\"processor_eq\\">
                        <option value=\\"\\">Select</option>
                        <option value=\\"intel\\">Intel</option>
                        <option value=\\"amd\\">AMD</option>
                    </select>
                </div>
                <div>
                    <label>Memory</label>
                    <select name=\\"memory_eq\\">
                        <option value=\\"\\">Select</option>
                        <option value=\\"8\\">8</option>
                        <option value=\\"16\\">16</option>
                        <option value=\\"32\\">32</option>
                    </select>
                </div>
                <div>
                    <label>Frequency Min</label>
                    <input
                        type=\\"number\\"
                        name=\\"frequency_gte\\"
                        min=\\"1\\"
                        max=\\"5\\"
                        value=\\"\\"
                    />
                </div>
                <div>
                    <label>Frequency Max</label>
                    <input
                        type=\\"number\\"
                        name=\\"frequency_lte\\"
                        min=\\"1\\"
                        max=\\"5\\"
                        value=\\"\\"
                    />
                </div>
            </form>
        </div>
        <div class=\\"col-6\\">
            <div class=\\"result\\">
                <ul>
                    <li>v1</li>
                    <li>d3</li>
                    <li>d2</li>
                </ul>
            </div>
        </div>
    </div>
</div>
"
`;

exports[`application 4`] = `
"<div class=\\"container m-3\\">
    <div class=\\"row\\">
        <div class=\\"col-6\\">
            <form>
                <div>
                    <label>Processor</label>
                    <select name=\\"processor_eq\\">
                        <option value=\\"\\">Select</option>
                        <option value=\\"intel\\">Intel</option>
                        <option value=\\"amd\\">AMD</option>
                    </select>
                </div>
                <div>
                    <label>Memory</label>
                    <select name=\\"memory_eq\\">
                        <option value=\\"\\">Select</option>
                        <option value=\\"8\\">8</option>
                        <option value=\\"16\\">16</option>
                        <option value=\\"32\\">32</option>
                    </select>
                </div>
                <div>
                    <label>Frequency Min</label>
                    <input
                        type=\\"number\\"
                        name=\\"frequency_gte\\"
                        min=\\"1\\"
                        max=\\"5\\"
                        value=\\"4\\"
                    />
                </div>
                <div>
                    <label>Frequency Max</label>
                    <input
                        type=\\"number\\"
                        name=\\"frequency_lte\\"
                        min=\\"1\\"
                        max=\\"5\\"
                        value=\\"\\"
                    />
                </div>
            </form>
        </div>
        <div class=\\"col-6\\">
            <div class=\\"result\\"></div>
        </div>
    </div>
</div>
"
`;

exports[`application 5`] = `
"<div class=\\"container m-3\\">
    <div class=\\"row\\">
        <div class=\\"col-6\\">
            <form>
                <div>
                    <label>Processor</label>
                    <select name=\\"processor_eq\\">
                        <option value=\\"\\">Select</option>
                        <option value=\\"intel\\">Intel</option>
                        <option value=\\"amd\\">AMD</option>
                    </select>
                </div>
                <div>
                    <label>Memory</label>
                    <select name=\\"memory_eq\\">
                        <option value=\\"\\">Select</option>
                        <option value=\\"8\\">8</option>
                        <option value=\\"16\\">16</option>
                        <option value=\\"32\\">32</option>
                    </select>
                </div>
                <div>
                    <label>Frequency Min</label>
                    <input
                        type=\\"number\\"
                        name=\\"frequency_gte\\"
                        min=\\"1\\"
                        max=\\"5\\"
                        value=\\"1\\"
                    />
                </div>
                <div>
                    <label>Frequency Max</label>
                    <input
                        type=\\"number\\"
                        name=\\"frequency_lte\\"
                        min=\\"1\\"
                        max=\\"5\\"
                        value=\\"\\"
                    />
                </div>
            </form>
        </div>
        <div class=\\"col-6\\">
            <div class=\\"result\\">
                <ul>
                    <li>v1</li>
                    <li>d3</li>
                    <li>d2</li>
                </ul>
            </div>
        </div>
    </div>
</div>
"
`;

exports[`application 6`] = `
"<div class=\\"container m-3\\">
    <div class=\\"row\\">
        <div class=\\"col-6\\">
            <form>
                <div>
                    <label>Processor</label>
                    <select name=\\"processor_eq\\">
                        <option value=\\"\\">Select</option>
                        <option value=\\"intel\\">Intel</option>
                        <option value=\\"amd\\">AMD</option>
                    </select>
                </div>
                <div>
                    <label>Memory</label>
                    <select name=\\"memory_eq\\">
                        <option value=\\"\\">Select</option>
                        <option value=\\"8\\">8</option>
                        <option value=\\"16\\">16</option>
                        <option value=\\"32\\">32</option>
                    </select>
                </div>
                <div>
                    <label>Frequency Min</label>
                    <input
                        type=\\"number\\"
                        name=\\"frequency_gte\\"
                        min=\\"1\\"
                        max=\\"5\\"
                        value=\\"1\\"
                    />
                </div>
                <div>
                    <label>Frequency Max</label>
                    <input
                        type=\\"number\\"
                        name=\\"frequency_lte\\"
                        min=\\"1\\"
                        max=\\"5\\"
                        value=\\"\\"
                    />
                </div>
            </form>
        </div>
        <div class=\\"col-6\\">
            <div class=\\"result\\">
                <ul>
                    <li>v1</li>
                    <li>d3</li>
                    <li>d2</li>
                </ul>
            </div>
        </div>
    </div>
</div>
"
`;

exports[`application 7`] = `
"<div class=\\"container m-3\\">
    <div class=\\"row\\">
        <div class=\\"col-6\\">
            <form>
                <div>
                    <label>Processor</label>
                    <select name=\\"processor_eq\\">
                        <option value=\\"\\">Select</option>
                        <option value=\\"intel\\">Intel</option>
                        <option value=\\"amd\\">AMD</option>
                    </select>
                </div>
                <div>
                    <label>Memory</label>
                    <select name=\\"memory_eq\\">
                        <option value=\\"\\">Select</option>
                        <option value=\\"8\\">8</option>
                        <option value=\\"16\\">16</option>
                        <option value=\\"32\\">32</option>
                    </select>
                </div>
                <div>
                    <label>Frequency Min</label>
                    <input
                        type=\\"number\\"
                        name=\\"frequency_gte\\"
                        min=\\"1\\"
                        max=\\"5\\"
                        value=\\"1\\"
                    />
                </div>
                <div>
                    <label>Frequency Max</label>
                    <input
                        type=\\"number\\"
                        name=\\"frequency_lte\\"
                        min=\\"1\\"
                        max=\\"5\\"
                        value=\\"2\\"
                    />
                </div>
            </form>
        </div>
        <div class=\\"col-6\\">
            <div class=\\"result\\">
                <ul>
                    <li>v1</li>
                </ul>
            </div>
        </div>
    </div>
</div>
"
`;
